src/topology/maps.lean

```diff
@@ -13,133 +13,203 @@ local attribute [instance] classical.prop_decidable
 
 variables {α : Type*} {β : Type*} {γ : Type*} {δ : Type*}
 
+section dense_range
+variables [topological_space α] [topological_space β] [topological_space γ]
+          (f : α → β) (g : β → γ)
+
+def dense_range := ∀ x, x ∈ closure (range f)
+
+lemma dense_range_iff_closure_eq : dense_range f ↔ closure (range f) = univ :=
+eq_univ_iff_forall.symm
+
+variables {f}
+
+lemma dense_range.comp (hg : dense_range g) (hf : dense_range f) (cg : continuous g) :
+  dense_range (g ∘ f) :=
+begin
+  have : g '' (closure $ range f) ⊆ closure (g '' range f),
+    from image_closure_subset_closure_image cg,
+  have : closure (g '' closure (range f)) ⊆ closure (g '' range f),
+    by simpa [closure_closure] using (closure_mono this),
+  intro c,
+  rw range_comp,
+  apply this,
+  rw [(dense_range_iff_closure_eq f).1 hf, image_univ],
+  exact hg c
+end
+
+lemma dense_range.inhabited (df : dense_range f) (b : β) : inhabited α :=
+⟨begin
+  have := exists_mem_of_ne_empty (mem_closure_iff.1 (df b) _ is_open_univ trivial),
+  simp only [mem_range, univ_inter] at this,
+  exact classical.some (classical.some_spec this),
+ end⟩
+
+end dense_range
+
+section inducing
+structure inducing [tα : topological_space α] [tβ : topological_space β] (f : α → β) : Prop :=
+(induced : tα = tβ.induced f)
+
+variables [topological_space α] [topological_space β] [topological_space γ] [topological_space δ]
+
+lemma inducing_id : inducing (@id α) :=
+⟨induced_id.symm⟩
+
+lemma inducing_compose {f : α → β} {g : β → γ} (hg : inducing g) (hf : inducing f) :
```

Comment:
```suggestion
lemma inducing.comp {f : α → β} {g : β → γ} (hg : inducing g) (hf : inducing f) :
```

Reply:
There are more `compose` lemmas in this area of mathlib. Maybe we should rename them all in one go in a separate PR.

Reply:
I fixed three of them.